### ISSUE TITLE
NoMethodError: undefined method `size' for nil:NilClass

### ISSUE THREAD

**Issue by devifr** [rubygems/rubygems#5981]
Fetching gem metadata from https://rubygems.org/..........
Resolving dependencies...
Using rake 13.0.6
Using concurrent-ruby 1.1.10
Using minitest 5.16.3
Using zeitwerk 2.6.1
Using thread_safe 0.3.6
Using builder 3.2.4
Using racc 1.6.0
Using erubi 1.11.0
Using crass 1.0.6
Using rack 2.2.4
Using nio4r 2.5.8
Using websocket-extensions 0.1.5
Using mini_mime 1.1.2
Using public_suffix 5.0.0
Using ansi 1.5.0
Using ast 2.4.2
Using bindex 0.8.1
Using msgpack 1.6.0
Using bundler 2.3.23
Using byebug 11.1.3
Using matrix 0.4.2
Using regexp_parser 2.6.0
Using childprocess 4.1.0
Using marcel 1.0.2
Using ruby-rc4 0.1.5
Using ffi 1.15.5
Using rb-fsevent 0.11.2
Using mysql2 0.4.10
Using method_source 1.0.0
Using pdfkit 0.8.7
Using ttfunk 1.7.0
Using thor 1.2.1
Using rexml 3.2.5
Using rubyzip 2.3.2
Using pdf-core 0.9.0
Using tilt 2.0.11
Using websocket 1.2.9
Using spring 2.1.1
Using turbolinks-source 5.2.0
Using i18n 1.12.0
Using tzinfo 1.2.10
Using nokogiri 1.13.8 (x86_64-linux)
Using rack-test 2.0.2
Using rack-proxy 0.7.4
Using sprockets 4.1.1
Using websocket-driver 0.7.5
Using mail 2.7.1
Using addressable 2.8.1
Using bootsnap 1.13.0
Using ruby-ll 2.1.2
Using puma 4.3.12
Using combine_pdf 1.0.22
Using rb-inotify 0.10.1
Using sassc 2.4.0
Using prawn 2.4.0
Using selenium-webdriver 4.5.0
Using turbolinks 5.2.1
Using activesupport 6.0.6
Using loofah 2.19.0
Using xpath 3.2.0
Using oga 3.4
Using grover 1.1.2
Using listen 3.7.1
Using prawn-table 0.2.2
Using rails-dom-testing 2.0.3
Using rails-html-sanitizer 1.4.3
Using globalid 1.0.0
Using activemodel 6.0.6
Using webdrivers 5.2.0
Using wicked_pdf 2.6.3
Using capybara 3.37.1
Using prawn-html 0.7.1
Using spring-watcher-listen 2.0.1
Using actionview 6.0.6
Using activejob 6.0.6
Using activerecord 6.0.6
Using actionpack 6.0.6
Using jbuilder 2.11.5
Using actioncable 6.0.6
Using activestorage 6.0.6
Using actionmailer 6.0.6
Using railties 6.0.6
Using sprockets-rails 3.4.2
Using actionmailbox 6.0.6
Using actiontext 6.0.6
Using sassc-rails 2.1.2
Using web-console 4.2.0
Using webpacker 4.3.0
Using rails 6.0.6
Using sass-rails 6.0.0
Exception while verifying /usr/local/lib/ruby/gems/2.7.0/cache/wkhtmltopdf-binary-0.12.6.5.gem
--- ERROR REPORT TEMPLATE -------------------------------------------------------

```
NoMethodError: undefined method `size' for nil:NilClass
  /usr/local/lib/ruby/site_ruby/2.7.0/rubygems/package/tar_reader/entry.rb:139:in `read'
  /usr/local/lib/ruby/site_ruby/2.7.0/rubygems/package.rb:365:in `block in digest'
  /usr/local/lib/ruby/site_ruby/2.7.0/rubygems/package.rb:362:in `each'
  /usr/local/lib/ruby/site_ruby/2.7.0/rubygems/package.rb:362:in `digest'
  /usr/local/lib/ruby/site_ruby/2.7.0/rubygems/package.rb:664:in `verify_entry'
  /usr/local/lib/ruby/site_ruby/2.7.0/rubygems/package.rb:683:in `block in verify_files'
  /usr/local/lib/ruby/site_ruby/2.7.0/rubygems/package/tar_reader.rb:65:in `each'
  /usr/local/lib/ruby/site_ruby/2.7.0/rubygems/package.rb:682:in `verify_files'
  /usr/local/lib/ruby/site_ruby/2.7.0/rubygems/package.rb:613:in `block (2 levels) in verify'
  /usr/local/lib/ruby/site_ruby/2.7.0/rubygems/package/tar_reader.rb:27:in `new'
  /usr/local/lib/ruby/site_ruby/2.7.0/rubygems/package.rb:610:in `block in verify'
  /usr/local/lib/ruby/site_ruby/2.7.0/rubygems/package/file_source.rb:29:in `open'
  /usr/local/lib/ruby/site_ruby/2.7.0/rubygems/package/file_source.rb:29:in `with_read_io'
  /usr/local/lib/ruby/site_ruby/2.7.0/rubygems/package.rb:609:in `verify'
  /usr/local/lib/ruby/site_ruby/2.7.0/rubygems/package.rb:590:in `spec'
  /usr/local/lib/ruby/site_ruby/2.7.0/rubygems/installer.rb:287:in `spec'
  /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/rubygems_gem_installer.rb:97:in `spec'
  /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/source/rubygems.rb:190:in `install'
  /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/installer/gem_installer.rb:54:in `install'
  /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/installer/gem_installer.rb:16:in `install_from_spec'
  /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/installer/parallel_installer.rb:186:in `do_install'
  /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/installer/parallel_installer.rb:177:in `block in worker_pool'
  /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/worker.rb:62:in `apply_func'
  /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/worker.rb:57:in `block in process_queue'
  /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/worker.rb:54:in `loop'
  /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/worker.rb:54:in `process_queue'
  /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.3.23/lib/bundler/worker.rb:91:in `block (2 levels) in create_threads'

```

## Environment

```
Bundler       2.3.23
  Platforms   ruby, x86_64-linux
Ruby          2.7.6p219 (2022-04-12 revision c9c2245c0a25176072e02db9254f0e0c84c805cd) [x86_64-linux]
  Full Path   /usr/local/bin/ruby
  Config Dir  /usr/local/etc
RubyGems      3.3.23
  Gem Home    /usr/local/lib/ruby/gems/2.7.0
  Gem Path    /root/.gem/ruby/2.7.0:/usr/local/lib/ruby/gems/2.7.0
  User Home   /root
  User Path   /root/.gem/ruby/2.7.0
  Bin Dir     /usr/local/bin
OpenSSL       
  Compiled    OpenSSL 1.0.1f 6 Jan 2014
  Loaded      OpenSSL 1.0.1f 6 Jan 2014
  Cert File   /usr/lib/ssl/cert.pem
  Cert Dir    /usr/lib/ssl/certs
Tools         
  Git         not installed
  RVM         not installed
  rbenv       not installed
  chruby      not installed
```

## Bundler Build Metadata

```
Built At          2022-10-05
Git SHA           250d9d485d
Released Version  true
```

### Gemfile

```ruby
source 'https://rubygems.org'
git_source(:github) { |repo| "https://github.com/#{repo}.git" }

ruby '2.7.6'

# Bundle edge Rails instead: gem 'rails', github: 'rails/rails', branch: 'main'
gem 'rails', '~> 6.0.4', '>= 6.0.4.7'
# Use sqlite3 as the database for Active Record
# gem 'sqlite3', '~> 1.4'
# Use Puma as the app server
gem 'puma', '~> 4.1'
# Use SCSS for stylesheets
gem 'sass-rails', '>= 6'
# Transpile app-like JavaScript. Read more: https://github.com/rails/webpacker
gem 'webpacker', '~> 4.0'
# Turbolinks makes navigating your web application faster. Read more: https://github.com/turbolinks/turbolinks
gem 'turbolinks', '~> 5'
# Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
gem 'jbuilder', '~> 2.7'
# Use Redis adapter to run Action Cable in production
# gem 'redis', '~> 4.0'
# Use Active Model has_secure_password
# gem 'bcrypt', '~> 3.1.7'
gem 'grover'
gem 'mysql2', '>= 0.3.18', '< 0.5'
gem "wicked_pdf"
gem "wkhtmltopdf-binary"
gem "pdfkit"
gem "prawn"
gem "prawn-table"
gem "prawn-html"


# Use Active Storage variant
# gem 'image_processing', '~> 1.2'

# Reduces boot times through caching; required in config/boot.rb
gem 'bootsnap', '>= 1.4.2', require: false

group :development, :test do
  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
  gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
end

group :development do
  # Access an interactive console on exception pages or by calling 'console' anywhere in the code.
  gem 'web-console', '>= 3.3.0'
  gem 'listen', '~> 3.2'
  # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
  gem 'spring'
  gem 'spring-watcher-listen', '~> 2.0.0'
end

group :test do
  # Adds support for Capybara system testing and selenium driver
  gem 'capybara', '>= 2.15'
  gem 'selenium-webdriver'
  # Easy installation and use of web drivers to run system tests with browsers
  gem 'webdrivers'
end

# Windows does not include zoneinfo files, so bundle the tzinfo-data gem
gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
```

### Gemfile.lock

```
GEM
  remote: https://rubygems.org/
  specs:
    actioncable (6.0.6)
      actionpack (= 6.0.6)
      nio4r (~> 2.0)
      websocket-driver (>= 0.6.1)
    actionmailbox (6.0.6)
      actionpack (= 6.0.6)
      activejob (= 6.0.6)
      activerecord (= 6.0.6)
      activestorage (= 6.0.6)
      activesupport (= 6.0.6)
      mail (>= 2.7.1)
    actionmailer (6.0.6)
      actionpack (= 6.0.6)
      actionview (= 6.0.6)
      activejob (= 6.0.6)
      mail (~> 2.5, >= 2.5.4)
      rails-dom-testing (~> 2.0)
    actionpack (6.0.6)
      actionview (= 6.0.6)
      activesupport (= 6.0.6)
      rack (~> 2.0, >= 2.0.8)
      rack-test (>= 0.6.3)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.2.0)
    actiontext (6.0.6)
      actionpack (= 6.0.6)
      activerecord (= 6.0.6)
      activestorage (= 6.0.6)
      activesupport (= 6.0.6)
      nokogiri (>= 1.8.5)
    actionview (6.0.6)
      activesupport (= 6.0.6)
      builder (~> 3.1)
      erubi (~> 1.4)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.1, >= 1.2.0)
    activejob (6.0.6)
      activesupport (= 6.0.6)
      globalid (>= 0.3.6)
    activemodel (6.0.6)
      activesupport (= 6.0.6)
    activerecord (6.0.6)
      activemodel (= 6.0.6)
      activesupport (= 6.0.6)
    activestorage (6.0.6)
      actionpack (= 6.0.6)
      activejob (= 6.0.6)
      activerecord (= 6.0.6)
      marcel (~> 1.0)
    activesupport (6.0.6)
      concurrent-ruby (~> 1.0, >= 1.0.2)
      i18n (>= 0.7, < 2)
      minitest (~> 5.1)
      tzinfo (~> 1.1)
      zeitwerk (~> 2.2, >= 2.2.2)
    addressable (2.8.1)
      public_suffix (>= 2.0.2, < 6.0)
    ansi (1.5.0)
    ast (2.4.2)
    bindex (0.8.1)
    bootsnap (1.13.0)
      msgpack (~> 1.2)
    builder (3.2.4)
    byebug (11.1.3)
    capybara (3.37.1)
      addressable
      matrix
      mini_mime (>= 0.1.3)
      nokogiri (~> 1.8)
      rack (>= 1.6.0)
      rack-test (>= 0.6.3)
      regexp_parser (>= 1.5, < 3.0)
      xpath (~> 3.2)
    childprocess (4.1.0)
    combine_pdf (1.0.22)
      matrix
      ruby-rc4 (>= 0.1.5)
    concurrent-ruby (1.1.10)
    crass (1.0.6)
    erubi (1.11.0)
    ffi (1.15.5)
    globalid (1.0.0)
      activesupport (>= 5.0)
    grover (1.1.2)
      combine_pdf (~> 1.0)
      nokogiri (~> 1.0)
    i18n (1.12.0)
      concurrent-ruby (~> 1.0)
    jbuilder (2.11.5)
      actionview (>= 5.0.0)
      activesupport (>= 5.0.0)
    listen (3.7.1)
      rb-fsevent (~> 0.10, >= 0.10.3)
      rb-inotify (~> 0.9, >= 0.9.10)
    loofah (2.19.0)
      crass (~> 1.0.2)
      nokogiri (>= 1.5.9)
    mail (2.7.1)
      mini_mime (>= 0.1.1)
    marcel (1.0.2)
    matrix (0.4.2)
    method_source (1.0.0)
    mini_mime (1.1.2)
    minitest (5.16.3)
    msgpack (1.6.0)
    mysql2 (0.4.10)
    nio4r (2.5.8)
    nokogiri (1.13.8-x86_64-linux)
      racc (~> 1.4)
    oga (3.4)
      ast
      ruby-ll (~> 2.1)
    pdf-core (0.9.0)
    pdfkit (0.8.7)
    prawn (2.4.0)
      pdf-core (~> 0.9.0)
      ttfunk (~> 1.7)
    prawn-html (0.7.1)
      oga (~> 3.3)
      prawn (~> 2.4)
    prawn-table (0.2.2)
      prawn (>= 1.3.0, < 3.0.0)
    public_suffix (5.0.0)
    puma (4.3.12)
      nio4r (~> 2.0)
    racc (1.6.0)
    rack (2.2.4)
    rack-proxy (0.7.4)
      rack
    rack-test (2.0.2)
      rack (>= 1.3)
    rails (6.0.6)
      actioncable (= 6.0.6)
      actionmailbox (= 6.0.6)
      actionmailer (= 6.0.6)
      actionpack (= 6.0.6)
      actiontext (= 6.0.6)
      actionview (= 6.0.6)
      activejob (= 6.0.6)
      activemodel (= 6.0.6)
      activerecord (= 6.0.6)
      activestorage (= 6.0.6)
      activesupport (= 6.0.6)
      bundler (>= 1.3.0)
      railties (= 6.0.6)
      sprockets-rails (>= 2.0.0)
    rails-dom-testing (2.0.3)
      activesupport (>= 4.2.0)
      nokogiri (>= 1.6)
    rails-html-sanitizer (1.4.3)
      loofah (~> 2.3)
    railties (6.0.6)
      actionpack (= 6.0.6)
      activesupport (= 6.0.6)
      method_source
      rake (>= 0.8.7)
      thor (>= 0.20.3, < 2.0)
    rake (13.0.6)
    rb-fsevent (0.11.2)
    rb-inotify (0.10.1)
      ffi (~> 1.0)
    regexp_parser (2.6.0)
    rexml (3.2.5)
    ruby-ll (2.1.2)
      ansi
      ast
    ruby-rc4 (0.1.5)
    rubyzip (2.3.2)
    sass-rails (6.0.0)
      sassc-rails (~> 2.1, >= 2.1.1)
    sassc (2.4.0)
      ffi (~> 1.9)
    sassc-rails (2.1.2)
      railties (>= 4.0.0)
      sassc (>= 2.0)
      sprockets (> 3.0)
      sprockets-rails
      tilt
    selenium-webdriver (4.5.0)
      childprocess (>= 0.5, < 5.0)
      rexml (~> 3.2, >= 3.2.5)
      rubyzip (>= 1.2.2, < 3.0)
      websocket (~> 1.0)
    spring (2.1.1)
    spring-watcher-listen (2.0.1)
      listen (>= 2.7, < 4.0)
      spring (>= 1.2, < 3.0)
    sprockets (4.1.1)
      concurrent-ruby (~> 1.0)
      rack (> 1, < 3)
    sprockets-rails (3.4.2)
      actionpack (>= 5.2)
      activesupport (>= 5.2)
      sprockets (>= 3.0.0)
    thor (1.2.1)
    thread_safe (0.3.6)
    tilt (2.0.11)
    ttfunk (1.7.0)
    turbolinks (5.2.1)
      turbolinks-source (~> 5.2)
    turbolinks-source (5.2.0)
    tzinfo (1.2.10)
      thread_safe (~> 0.1)
    web-console (4.2.0)
      actionview (>= 6.0.0)
      activemodel (>= 6.0.0)
      bindex (>= 0.4.0)
      railties (>= 6.0.0)
    webdrivers (5.2.0)
      nokogiri (~> 1.6)
      rubyzip (>= 1.3.0)
      selenium-webdriver (~> 4.0)
    webpacker (4.3.0)
      activesupport (>= 4.2)
      rack-proxy (>= 0.6.1)
      railties (>= 4.2)
    websocket (1.2.9)
    websocket-driver (0.7.5)
      websocket-extensions (>= 0.1.0)
    websocket-extensions (0.1.5)
    wicked_pdf (2.6.3)
      activesupport
    xpath (3.2.0)
      nokogiri (~> 1.8)
    zeitwerk (2.6.1)

PLATFORMS
  x86_64-linux

DEPENDENCIES
  bootsnap (>= 1.4.2)
  byebug
  capybara (>= 2.15)
  grover
  jbuilder (~> 2.7)
  listen (~> 3.2)
  mysql2 (>= 0.3.18, < 0.5)
  pdfkit
  prawn
  prawn-html
  prawn-table
  puma (~> 4.1)
  rails (~> 6.0.4, >= 6.0.4.7)
  sass-rails (>= 6)
  selenium-webdriver
  spring
  spring-watcher-listen (~> 2.0.0)
  turbolinks (~> 5)
  tzinfo-data
  web-console (>= 3.3.0)
  webdrivers
  webpacker (~> 4.0)
  wicked_pdf

RUBY VERSION
   ruby 2.7.6p219

BUNDLED WITH
   2.3.23
```

--- TEMPLATE END ----------------------------------------------------------------

**Comment by pgundlupetvenkatesh** [comment]
I am getting the same with `NoMethodError: undefined method 'full_name' for nil:NilClass`

Ruby v2.6.3p62 (2019-04-16 revision 67580) [x64-mingw32]

```
$ bundle update
Fetching https://github.com/bugsnag/yard-cucumber.git
Fetching gem metadata from <artifactory_source>....
Fetching gem metadata from https://rubygems.org/...........
Resolving dependencies...
Your lockfile was created by an old Bundler that left some things out.
Because of the missing DEPENDENCIES, we can only install gems one at a time, instead of installing 8 at a time.
You can fix this by adding the missing gems to your Gemfile, running bundle install, and then removing the gems from your Gemfile.
The missing gems are:
* mini_portile2 depended upon by nokogiri
Using rake 13.0.6
Using concurrent-ruby 1.1.10
Using i18n 1.12.0
Using minitest 5.16.3
Using tzinfo 2.0.5
Using zeitwerk 2.6.1
Using activesupport 6.1.7
Using builder 3.2.4
Using erubi 1.11.0
Using racc 1.6.0
Installing nokogiri 1.13.8 with native extensions
```
--- ERROR REPORT TEMPLATE -------------------------------------------------------

```
NoMethodError: undefined method `full_name' for nil:NilClass
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/installer/parallel_installer.rb:124:in `block (2 levels) in check_for_unmet_dependencies'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/installer/parallel_installer.rb:123:in `each'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/installer/parallel_installer.rb:123:in `block in check_for_unmet_dependencies'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/installer/parallel_installer.rb:122:in `each'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/installer/parallel_installer.rb:122:in `check_for_unmet_dependencies'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/installer/parallel_installer.rb:100:in `call'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/installer/parallel_installer.rb:71:in `call'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/installer.rb:254:in `install_in_parallel'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/installer.rb:209:in `install'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/installer.rb:89:in `block in run'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/process_lock.rb:12:in `block in lock'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/process_lock.rb:9:in `open'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/process_lock.rb:9:in `lock'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/installer.rb:71:in `run'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/installer.rb:23:in `install'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/cli/update.rb:78:in `run'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/cli.rb:305:in `block in update'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/settings.rb:131:in `temporary'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/cli.rb:304:in `update'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/cli.rb:31:in `dispatch'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/cli.rb:25:in `start'
  <dir_path>/Ruby26-x64/lib/ruby/gems/2.6.0/gems/bundler-2.3.23/exe/bundle:48:in `block in <top (required)>'
  <dir_path>/Ruby26-x64/lib/ruby/site_ruby/2.6.0/bundler/friendly_errors.rb:120:in `with_friendly_errors'
  <dir_path>/Ruby26-x64/lib/ruby/gems/2.6.0/gems/bundler-2.3.23/exe/bundle:36:in `<top (required)>'
  <dir_path>/Ruby26-x64/bin/bundle:32:in `load'
  <dir_path>/Ruby26-x64/bin/bundle:32:in `<main>'

```

## Environment

```
Bundler       2.3.23
  Platforms   ruby, x64-mingw32
Ruby          2.6.3p62 (2019-04-16 revision 67580) [x64-mingw32]
  Full Path   <dir_path>/Ruby26-x64/bin/ruby.exe
  Config Dir  <ProgramData_path>
RubyGems      3.3.23
  Gem Home    <dir_path>/Ruby26-x64/lib/ruby/gems/2.6.0
  Gem Path    <user_path>/.gem/ruby/2.6.0;<dir_path>/Ruby26-x64/lib/ruby/gems/2.6.0
  User Home   <user_path>
  User Path   <user_path>/.gem/ruby/2.6.0
  Bin Dir     <dir_path>/Ruby26-x64/bin
OpenSSL
  Compiled    OpenSSL 1.1.1b  26 Feb 2019
  Loaded      OpenSSL 1.1.1b  26 Feb 2019
  Cert File   <dir_path>/Ruby26-x64/ssl/cert.pem
  Cert Dir    <dir_path>/Ruby26-x64/ssl/certs
Tools
  Git         2.35.1.windows.2
  RVM         not installed
  rbenv       not installed
  chruby      not installed
```

## Bundler Build Metadata

```
Built At          2022-10-05
Git SHA           250d9d485d
Released Version  true
```

## Gemfile

### Gemfile

```ruby
source 'https://rubygems.org'

source '<artifactory_source>' do
  # loading loacal gems
end

# Latest version of Bundler
gem 'bundler', '~> 2.0'

# Rake Tasks
gem 'rake', '~> 13.0'

# Manage SSL Certificates for HTTPS connections to GitHub and RubyGems
gem 'certified', '~> 1.0'

# Documentation
gem 'yard', '~> 0.9.0' # Docs
gem 'yard-cucumber', :github => 'bugsnag/yard-cucumber'
gem 'titleize', '~> 1.0' # https://github.com/granth/titleize
gem 'wannabe_bool', '~> 0.7.1' # Adds #to_b method

# Test Runner + Assertions
gem 'webdrivers', '~> 5.0.0' # Selenium WebDriver downloader and updater
gem 'cucumber', '~> 6.1.0' # Test Manager & Runner
gem 'minitest', '~> 5.11' # Asserts

# Fix for childprocess gem used by selenium-webdriver on Windows.
gem 'ffi'

# Rails Helper Methods
gem 'activesupport', '~> 6.0'
gem 'actionview', '~> 6.0'

# Debuggers
gem 'pry-byebug'
gem 'pry-awesome_print' # pry + awesome_print in one
```

### Gemfile.lock

```
GIT
  remote: https://github.com/bugsnag/yard-cucumber.git
  revision: 679d8a834152b53df4700b7107a71d0753e5a2ba
  specs:
    yard-cucumber (4.0.0)
      cucumber (>= 2.0, < 8.0)
      yard (~> 0.8, >= 0.8.1)

GEM
  remote: <artifactory_source>
  specs:
    # local gems list

GEM
  remote: https://rubygems.org/
  specs:
    actionview (6.1.7)
      activesupport (= 6.1.7)
      builder (~> 3.1)
      erubi (~> 1.4)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.1, >= 1.2.0)
    activesupport (6.1.7)
      concurrent-ruby (~> 1.0, >= 1.0.2)
      i18n (>= 1.6, < 2)
      minitest (>= 5.1)
      tzinfo (~> 2.0)
      zeitwerk (~> 2.3)
    addressable (2.8.1)
      public_suffix (>= 2.0.2, < 6.0)
    akami (1.3.1)
      gyoku (>= 0.4.0)
      nokogiri
    awesome_print (1.9.2)
    binding_of_caller (1.0.0)
      debug_inspector (>= 0.0.1)
    builder (3.2.4)
    byebug (11.1.3)
    certified (1.0.0)
    childprocess (4.1.0)
    clipboard (1.3.6)
    coderay (1.1.3)
    concurrent-ruby (1.1.10)
    crass (1.0.6)
    cucumber (6.1.0)
      builder (~> 3.2, >= 3.2.4)
      cucumber-core (~> 9.0, >= 9.0.1)
      cucumber-create-meta (~> 4.0, >= 4.0.0)
      cucumber-cucumber-expressions (~> 12.1, >= 12.1.1)
      cucumber-gherkin (~> 18.1, >= 18.1.0)
      cucumber-html-formatter (~> 13.0, >= 13.0.0)
      cucumber-messages (~> 15.0, >= 15.0.0)
      cucumber-wire (~> 5.0, >= 5.0.1)
      diff-lcs (~> 1.4, >= 1.4.4)
      mime-types (~> 3.3, >= 3.3.1)
      multi_test (~> 0.1, >= 0.1.2)
      sys-uname (~> 1.2, >= 1.2.2)
    cucumber-core (9.0.1)
      cucumber-gherkin (~> 18.1, >= 18.1.0)
      cucumber-messages (~> 15.0, >= 15.0.0)
      cucumber-tag-expressions (~> 3.0, >= 3.0.1)
    cucumber-create-meta (4.0.0)
      cucumber-messages (~> 15.0, >= 15.0.0)
      sys-uname (~> 1.2, >= 1.2.2)
    cucumber-cucumber-expressions (12.1.3)
    cucumber-gherkin (18.1.1)
      cucumber-messages (~> 15.0, >= 15.0.0)
    cucumber-html-formatter (13.0.0)
      cucumber-messages (~> 15.0, >= 15.0.0)
    cucumber-messages (15.0.0)
      protobuf-cucumber (~> 3.10, >= 3.10.8)
    cucumber-tag-expressions (3.0.1)
    cucumber-wire (5.0.1)
      cucumber-core (~> 9.0, >= 9.0.1)
      cucumber-cucumber-expressions (~> 12.1, >= 12.1.1)
      cucumber-messages (~> 15.0, >= 15.0.0)
    data_magic (1.2)
      faker (>= 1.1.2)
      yml_reader (>= 0.6)
    debug_inspector (1.1.0)
    diff-lcs (1.5.0)
    erubi (1.11.0)
    faker (2.22.0)
      i18n (>= 1.8.11, < 2)
    ffi (1.15.5-x64-mingw32)
    gyoku (1.4.0)
      builder (>= 2.1.2)
      rexml (~> 3.0)
    httpi (2.5.0)
      rack
      socksify
    i18n (1.12.0)
      concurrent-ruby (~> 1.0)
    loofah (2.19.0)
      crass (~> 1.0.2)
      nokogiri (>= 1.5.9)
    method_source (1.0.0)
    middleware (0.1.0)
    mime-types (3.4.1)
      mime-types-data (~> 3.2015)
    mime-types-data (3.2022.0105)
    mini_magick (4.11.0)
    minitest (5.16.3)
    multi_test (0.1.2)
    nokogiri (1.13.8-x64-mingw32)
      racc (~> 1.4)
    nori (2.6.0)
    os (1.1.4)
    page-object (2.3.1)
      page_navigation (>= 0.10)
      watir (>= 6.10.3)
    page_navigation (0.10)
      data_magic (>= 0.22)
    protobuf-cucumber (3.10.8)
      activesupport (>= 3.2)
      middleware
      thor
      thread_safe
    pry (0.13.1)
      coderay (~> 1.1)
      method_source (~> 1.0)
    pry-awesome_print (9.6.11)
      awesome_print (>= 1.1.0, < 999)
      pry (>= 0.9.0, < 999)
    pry-byebug (3.9.0)
      byebug (~> 11.0)
      pry (~> 0.13.0)
    public_suffix (5.0.0)
    racc (1.6.0)
    rack (3.0.0)
    rails-dom-testing (2.0.3)
      activesupport (>= 4.2.0)
      nokogiri (>= 1.6)
    rails-html-sanitizer (1.4.3)
      loofah (~> 2.3)
    rake (13.0.6)
    regexp_parser (2.6.0)
    rexml (3.2.5)
    ruby-oci8 (2.2.11-x64-mingw32)
    rubyzip (2.3.2)
    savon (2.12.1)
      akami (~> 1.2)
      builder (>= 2.1.2)
      gyoku (~> 1.2)
      httpi (~> 2.3)
      nokogiri (>= 1.8.1)
      nori (~> 2.4)
      wasabi (~> 3.4)
    selenium-webdriver (4.1.0)
      childprocess (>= 0.5, < 5.0)
      rexml (~> 3.2, >= 3.2.5)
      rubyzip (>= 1.2.2)
    socksify (1.7.1)
    sys-uname (1.2.2)
      ffi (~> 1.1)
    thor (1.2.1)
    thread_safe (0.3.6)
    titleize (1.4.1)
    tzinfo (2.0.5)
      concurrent-ruby (~> 1.0)
    wannabe_bool (0.7.1)
    wasabi (3.7.0)
      addressable
      httpi (~> 2.0)
      nokogiri (>= 1.4.2)
    watir (7.1.0)
      regexp_parser (>= 1.2, < 3)
      selenium-webdriver (~> 4.0)
    watir-screenshot-stitch (0.8.0)
      binding_of_caller (~> 1.0)
      mini_magick (~> 4.11)
      os (~> 1.1)
      rubyzip (~> 2.3)
      watir (~> 7.1)
    webdrivers (5.0.0)
      nokogiri (~> 1.6)
      rubyzip (>= 1.3.0)
      selenium-webdriver (~> 4.0)
    webrick (1.7.0)
    yard (0.9.28)
      webrick (~> 1.7.0)
    yml_reader (0.7)
    zeitwerk (2.6.0)

PLATFORMS
  x64-mingw32

DEPENDENCIES
  actionview (~> 6.0)
  activesupport (~> 6.0)
  bundler (~> 2.0)
  certified (~> 1.0)
  cucumber (~> 6.1.0)
  ffi
  minitest (~> 5.11)
  pry-awesome_print
  pry-byebug
  rake (~> 13.0)
  titleize (~> 1.0)
  wannabe_bool (~> 0.7.1)
  webdrivers (~> 5.0.0)
  yard (~> 0.9.0)
  yard-cucumber!

BUNDLED WITH
   2.3.18
```

--- TEMPLATE END ----------------------------------------------------------------

Unfortunately, an unexpected error occurred, and Bundler cannot continue.

First, try this link to see if there are any existing issue reports for this error:
https://github.com/rubygems/rubygems/search?q=undefined+method+%60full_name%27+for+nil+NilClass&type=Issues

If there aren't any reports for this error yet, please fill in the new issue form located at https://github.com/rubygems/rubygems/issues/new?labels=Bundler&template=bundler-related-issue.md, and copy and paste the report template above in there.
```

**Comment by deivid-rodriguez** [comment]
Hei @devifr!

Is this error reproducible every time?

**Comment by Opti-kjh** [comment]
> 헤이@devifr!
> 
> 이 오류가 매번 재현됩니까?

The same phenomenon is happening to me.

**Comment by coldnebo** [comment]
we just encountered the same error, but it wasn't for running out of disk space.  it is on NFS systems, are there other triggers for this error?  clearing the /cache directory and retrying solved our problem, but since it was a relatively new install we setup last week, I'd like to understand how the cache got corrupted so early?

Another colleague got a slightly different error:

```
--- ERROR REPORT TEMPLATE -------------------------------------------------------

ArgumentError: Malformed version number string 2.4.12
b
  /DONOTDELETE/shared/rvm/rubies/ruby-3.1.2/lib/ruby/site_ruby/3.1.0/rubygems/version.rb:223:in `initialize'
  /DONOTDELETE/shared/rvm/rubies/ruby-3.1.2/lib/ruby/site_ruby/3.1.0/rubygems/version.rb:206:in `new'
```

**Comment by deepak45gupta** [comment]
Hi can anyone help me with this, I am facing same issue I don't find this as memory storage issue as I can see Usage of /:   55.2% of 7.57GB it means half of the space is empty. 
@devifr did you find any solution?

**Comment by deepak45gupta** [comment]
> Hi can anyone help me with this, I am facing same issue I don't find this as memory storage issue as I can see Usage of /: 55.2% of 7.57GB it means half of the space is empty. @devifr did you find any solution?

I solved this issue by changing my gem from miniracer to therubyracer, I updated my gemfile and pushed it to version control, and then did, cap production deploy.

**Comment by andrey-lizunov** [comment]
Same thing and it's not connected with disk space. 
@deivid-rodriguez  Not sure why this issue was closed.

**Comment by deivid-rodriguez** [comment]
Because we fixed the particular instance of this error that we managed to reproduce.

Feel free to open a separate issue with reproducible steps if you're still running into the same error with the latest versions of RubyGems and Bundler.

Thanks!

**Comment by andrey-lizunov** [comment]
@deivid-rodriguez  Not sure, but seems that the reason is that the gem doesn't have release version 2.12.6.5, but at same time this version exists in rubygems.
![image](https://github.com/rubygems/rubygems/assets/1709731/c7c0ce8f-689a-4ae9-9470-1f6a3e726851)

**Comment by martinemde** [comment]
I think this should raise a Gem::Package::FormatError that states an unexpected EOF so that the error is more clear. Calling size on `nil` is not a great error regardless of the ultimate cause. I'll submit a PR for this in a moment.

**Comment by deivid-rodriguez** [comment]
Agreed @martinemde. That's a good improvement that will also probably make figuring the root cause easier 👍.

**Comment by martinemde** [comment]
#6882 fixes this by raising a more descriptive error on corrupt gem files. We can consider this truly closed now. Thanks for submitting this bug report.

**Comment by tovarish39** [comment]
У меня была такая ошибка на приложении rails 7.0.6. не понял причины ошибки, но получилось обойти таким путём:

- закомментил некоторые гемы в Gemfile.(начал с последних 3-х
  - gem "capybara"
  - gem "selenium-webdriver"
  - gem "webdrivers"
  )
- затем bundle install прошёл без ошибок
- затем по одному раскомментил и делал bundle install

в итоге все гемы установились.

**Comment by simi** [comment]
@tovarish39 strange that installing one by one fixed this. Have you been able to reproduce this?

**Comment by deivid-rodriguez** [comment]
Please upgrade rubygems (`gem update --system) if you run into this error. We should be giving a better error now after #6882.
